homeassistant/util/secrets.py

```diff
@@ -0,0 +1,133 @@
+"""Module to help keep passwords and other secrets."""
+
+import os
+import logging
+import collections
+from enum import Enum
+from homeassistant.config import load_yaml_config_file
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+try:
+    import keyring  # pylint: disable=wrong-import-order,wrong-import-position
```

Comment:
pylint complained about the try catch around the import if it is before the home assistant imports. Not sure if there's a better way to do it?

And keyring should probably be optional